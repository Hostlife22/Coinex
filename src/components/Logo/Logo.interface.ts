import { DetailedHTMLProps, HTMLAttributes } from 'react';

export interface ILogoProps extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
  handleMenu: () => void;
  isOpen: boolean;
}
