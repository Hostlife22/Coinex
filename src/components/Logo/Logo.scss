.logo {
  position: relative;
  width: 180px;
  transition: all 0.5s ease;

  &.logo_close {
    width: 48px;

    .logo__text {
      opacity: 0;
    }

    .logo__btn-toggle {
      transform: translateY(-50%) rotate(0deg);
    }
  }

  &__container {
    display: flex;
    align-items: center;
  }

  &__name {
    margin-left: -5px;
    font-weight: 600;
    font-size: 20px;
  }

  &__text {
    font-weight: 500;
    font-size: 17px;
    white-space: nowrap;
    opacity: 1;
    transition: all 0.3s ease;

    &_close {
      opacity: 0;
    }
  }

  &__image {
    display: flex;
    align-items: center;
    justify-content: center;

    img {
      width: 40px;
      border-radius: 6px;
    }
  }

  &__btn-toggle {
    position: absolute;
    top: 50%;
    right: -25px;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 25px;
    height: 25px;
    color: var(--color-white);
    font-size: 22px;
    background-color: var(--color-primary);
    border-radius: 50%;
    transform: translateY(-50%) rotate(180deg);
    cursor: pointer;
    transition: all 0.5s ease;

    &::after {
      position: absolute;
      top: 0;
      bottom: 0;
      width: 2px;
      background-color: red;
      content: '';
    }
  }
}

@media screen and (max-width: 1024px) {
  .logo {
    width: 150px;

    &__btn-toggle {
      right: -15px;
      width: 20px;
      height: 20px;
    }
  }
}

@media screen and (max-width: 520px) {
  .logo {
    &__image {
      width: 10px;
      height: 10px;
    }
  }
}
