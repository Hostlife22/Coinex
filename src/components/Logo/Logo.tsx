import cn from 'classnames';
import { FaArrowRight } from 'react-icons/fa';
import logo from '../../assets/logo.svg';
import useMediaQuery from '../../hooks/useMediaQuery';
import Ptag from '../Ptag/Ptag';
import { ILogoProps } from './Logo.interface';
import './Logo.scss';

const Logo = ({ handleMenu, isOpen, className, ...props }: ILogoProps) => {
  const matches = useMediaQuery('(max-width: 630px)');

  return (
    <div className={cn(className, 'logo', { logo_close: isOpen })} {...props}>
      <div className="logo__container">
        <span className="logo__image">
          <img src={logo} alt="logo" />
        </span>

        <div className="logo__text ">
          <Ptag className="logo__name">oineX</Ptag>
        </div>
      </div>

      {!matches && <FaArrowRight className="logo__btn-toggle" onClick={handleMenu} />}
    </div>
  );
};

export default Logo;
