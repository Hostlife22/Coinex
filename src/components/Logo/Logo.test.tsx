import { render, screen } from '@testing-library/react';
import MatchMediaMock from 'jest-matchmedia-mock';
import Logo from './Logo';

let matchMedia: MatchMediaMock;
const handleMenu = jest.fn();

describe('Logo component', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  it('renders logo', () => {
    const mediaQuery = '(min-width: 630px)';
    matchMedia.useMediaQuery(mediaQuery);

    render(<Logo isOpen={true} handleMenu={handleMenu} />);

    expect(screen.getByAltText('logo')).toBeInTheDocument();
    expect(screen.getByText(/oineX/i)).toBeInTheDocument();
    expect(screen.getByTestId('toggle')).toBeInTheDocument();
  });

  it('renders logo with custom className', () => {
    const mediaQuery = '(max-width: 630px)';
    matchMedia.useMediaQuery(mediaQuery);

    render(<Logo isOpen={true} handleMenu={handleMenu} />);

    const logo = screen.getByTestId('logo');

    expect(logo).toBeDefined();
    expect(logo).toHaveClass('logo_close');
  });

  it('сheck for adding and removing the class', () => {
    const mediaQuery = '(min-width: 630px)';
    matchMedia.useMediaQuery(mediaQuery);

    const { rerender } = render(<Logo isOpen={true} handleMenu={handleMenu} />);
    const toogle = screen.getByTestId('toggle');

    expect(toogle).toBeDefined();
    expect(screen.getByTestId('logo')).toHaveClass('logo_close');

    rerender(<Logo isOpen={false} handleMenu={handleMenu} />);
    expect(screen.getByTestId('logo')).not.toHaveClass('logo_close');
  });
});
