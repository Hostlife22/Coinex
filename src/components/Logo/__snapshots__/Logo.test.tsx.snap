// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component input snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="logo logo_close"
        data-testid="logo"
      >
        <div
          class="logo__container"
        >
          <span
            class="logo__image"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
          </span>
          <div
            class="logo__text "
          >
            <p
              class="logo__name"
              data-testid="logo-title"
            >
              oineX
            </p>
          </div>
        </div>
        <svg
          class="logo__btn-toggle"
          data-testid="toggle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="logo logo_close"
      data-testid="logo"
    >
      <div
        class="logo__container"
      >
        <span
          class="logo__image"
        >
          <img
            alt="logo"
            src="logo.svg"
          />
        </span>
        <div
          class="logo__text "
        >
          <p
            class="logo__name"
            data-testid="logo-title"
          >
            oineX
          </p>
        </div>
      </div>
      <svg
        class="logo__btn-toggle"
        data-testid="toggle"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Logo component logo snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="logo logo_close"
        data-testid="logo"
      >
        <div
          class="logo__container"
        >
          <span
            class="logo__image"
          >
            <img
              alt="logo"
              src="logo.svg"
            />
          </span>
          <div
            class="logo__text "
          >
            <p
              class="logo__name"
              data-testid="logo-title"
            >
              oineX
            </p>
          </div>
        </div>
        <svg
          class="logo__btn-toggle"
          data-testid="toggle"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
          />
        </svg>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="logo logo_close"
      data-testid="logo"
    >
      <div
        class="logo__container"
      >
        <span
          class="logo__image"
        >
          <img
            alt="logo"
            src="logo.svg"
          />
        </span>
        <div
          class="logo__text "
        >
          <p
            class="logo__name"
            data-testid="logo-title"
          >
            oineX
          </p>
        </div>
      </div>
      <svg
        class="logo__btn-toggle"
        data-testid="toggle"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z"
        />
      </svg>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
